Arduino/motor_control/drive.cpp

```diff
@@ -0,0 +1,251 @@
+#include "drive.h"
+
+// ============== Public Methods ==============
+
+// Left right orientation based on looking at the robot from the front position
+Drive::Drive(int pwm_pin_left, int pwm_pin_right, int dir_pin_left, int dir_pin_right){
+  pinMode(pwm_pin_left, OUTPUT);
+  pinMode(pwm_pin_right, OUTPUT);
+  pinMode(dir_pin_left, OUTPUT);
+  pinMode(dir_pin_right, OUTPUT);
+
+  _pwm_pin_left = pwm_pin_left;
+  _pwm_pin_right = pwm_pin_right;
+  _dir_pin_left = dir_pin_left;
+  _dir_pin_right = dir_pin_right;
+
+  // Initialize
+  _current_speed = 0; 
+  _current_rpm = 0;
+  _mode = STOPPED; 
+
+  // Ensure motor is initially stopped
+  analogWrite(_pwm_pin_left, 254);
+  analogWrite(_pwm_pin_right, 254);
+}
+
+/**
+* input parameter 1 from accel_rate enum, e.g. "ACC_LOW"
+* input parameter 2 optional, defaults to true (forward)
+*
+* Currently, accelerate only operates in the forward direction
+* Accelerates to maximun speed from the current speed
+*/
+void Drive::accelerate(int rate, bool fwd = true){
+
+	Serial.println("==> Accelerating");
+
+	// Set robot state
+	_mode = ACCELERATING; 
+
+	/// Set direction of motors
+  fwd ? _set_forward() : _set_backward();
+
+	// Get current speed as starting point
+	int curr_speed_i = convert_speed_to_i(_current_speed); 
+
+	for(int i=curr_speed_i; i>0; i--){
+
+		analogWrite(_pwm_pin_left, i);
+		analogWrite(_pwm_pin_right, i);
+		Serial.print("Speed up: ");
+		Serial.println(i);
+
+		// Track new current speeds
+		_current_speed = convert_i_to_speed(i); 
+		_current_rpm = convert_i_to_rpm(i); 
+
+		delay(5+50*(1-rate/6)); //vary scalar multiple to adjust minimum acceleration
+	}
+
+	// Set new speed 
+	_current_speed = MAX_SPEED; 
+	_current_rpm = MAX_RPM; 
+}
+
+/**
+* input parameter from accel_rate enum, e.g. "ACC_LOW"
+* input parameter 2 optional, defaults to true (forward)
+* 
+* Doesn't bring robot to a complete speed, only to a minimun speed from current position
+*/
+void Drive::deccelerate(int rate, bool fwd = true){
```

Comment:
For decelerate should the forward enum be default true? Not sure what the right course of action would be

Reply:
I think it makes sense to be default true because in 99% of cases we'll be decelerating when moving forward.